doc/security.rst

```diff
@@ -0,0 +1,184 @@
+machines would increase the damage resulting from leaking that key.
+
+Currently, `LUKS`_ is used for the rootfs partition. Using LUKS has
+several advantages:
+
+* ``refkit-initramfs`` can detect whether a partition is encrypted
+
+* LUKS tools have sane default parameters for encryption
+
+* encryption parameters are stored in the partition
+  instead of having to be hard-coded in the installer and
+  initramfs, which also simplifies updates (initramfs will
+  always use the parameters that were used when creating
+  the partition)
+
+The downside is a higher risk for data loss, because the LUKS header
+blocks are required for mounting the partition. This could be avoided
+by setting up ``dm-crypt`` directly, which currently is not supported
+out-of-the-box.
+
+.. _LUKS: https://gitlab.com/cryptsetup/cryptsetup/blob/master/README.md
+
+To unlock the LUKS partition, a secret key is
+needed. ``image-installer`` and the corresponding
+:file:`meta-refkit/recipes-image/images/initramfs-framework-refkit-luks.bb`
+can use a fixed passphrase. To use that, invoke image-installer with::
+
+  FIXED_PASSWORD=yes image-installer
+
+Obviously this is not secure and only useful for testing. Something
+that may be secure enough for real products (depending on the threat
+model for the product) is storing the secret key in a TPM chip. At the
+moment, TPM 1.2 is supported. If ``image-installer`` finds a TPM
+device, it will take over ownership of the TPM and use it to store a
+randomly generated key in a fixed NVRAM slot. The assumption is that the
+new OS installation has full control over the TPM and does not need to
+share it with other OS installations.
+
+When booting, the initramfs will retrieve the key and then prevent
+further access to it, using the TPM ``READ_STCLEAR`` protection
+feature for this particular slot. The effect is that even if an
+attacker gains access to the TPM at runtime, reading the key again
+gets denied by the TPM chip itself. However, the secret key is of
+course still in use by the kernel and thus might be leaked from there
+and/or files modified at runtime by an attacker will be written to
+disk permanently.
+
+Secure Boot
+===========
+
+The approach taken in images derived from :file:`refkit-image.bbclass`
+is to build a single UEFI application, the so called "UEFI combo
+application". This application contains:
+
+* systemd-boot (formerly known as gummiboot) EFI stub
+
+* Linux kernel
+
+* initramfs
+
+* fixed boot parameters
+
+* Runtime Machine Configuration (RMC) database with additional
+  boot parameters which get activated depending on the current machine
+
+This UEFI combo application gets loaded directly by the firmware,
+without any intermediate boot loader involved. This approach is fast,
+simple and can be secured by signing the UEFI combo application.
+
+That signing currently has to be done manually as described in the
+`Ostro(TM) OS Secure Boot`_ documentation.
```

Comment:
Perhaps. @mythi, are you working on enhancing support for Secure Boot in refkit?